src/exec/builtins/functions.rs

```diff
@@ -84,26 +99,163 @@ pub fn list_directory(shell: &mut ShellState, args: Vec<&str>) -> Result<()> {
+        
+        username_len_last = path.owner().unwrap().to_string().len();
+
+        if username_len_last > username_len {
+            username_len = username_len_last;
+        }
+    }
+
+    for i in &item {
+        let path = path_to_read.join(i);
+        let permission_octal =  { 
+            let x = format!("{:o}", std::fs::metadata(path.to_path_buf()).unwrap().permissions().mode()); 
+            x[x.len() - 3..].to_string() 
+        };
+
+        let permissions = match permission_format {
+            DirectoryListPermissionMode::Octal => {
+                permission_octal.white()
+            },
+                
+            DirectoryListPermissionMode::String => {
+                let permission_str = permission_octal;
+                let mut result = String::new();
+
+                let dash = "-".dark_grey();
+                let r = "r".white();
+                let w = "w".white();
+                let x = "x".white();
+
+                let mapping = vec![
+                    format!("{0}{0}{0}", dash),
+                    format!("{0}{0}{1}", dash, x), 
+                    format!("{0}{1}{0}", dash, w), 
+                    format!("{}{}{}", dash, w, x),
+                    format!("{}{1}{1}", r, dash),
+                    format!("{}{}{}", r, dash, x),
+                    format!("{}{}{}", r, w, dash),
+                    format!("{}{}{}", r, w, x)
+                ];
+
+                for (i, c) in permission_str.chars().enumerate() {
+                    let digit = c.to_digit(8).unwrap() as usize;
+
+                    if permission_seperator {
+                        if i == 0 {
+                            result.push_str(format!("{} ", "U".grey()).as_str());
+                        }
+
+                        if i == 1 {
+                            result.push_str(format!(" {} ", "G".grey()).as_str());
+                        }
+
+                        if i == 2 {
+                            result.push_str(format!(" {} ", "O".grey()).as_str());
+                        }
+                    }
+
+                    result.push_str(&mapping[digit]);
+                }
+
+                result.white()
+            },
+        };
+
+        let file_size = std::fs::metadata(path.to_path_buf()).unwrap().size();
+        let formatted_fsize = Size::from_bytes(file_size);
+
+        println!("{4} {3} {2} {1} {0}", 
+            if i.starts_with('.') { 
+                i.as_str().dark_grey() 
+            } else if i.chars().last().unwrap() == '/' {
+                if i.starts_with('.') {
+                    i.as_str().dark_green()
+                } else {
+                    i.as_str().green() 
+                }
+            } else { 
+                i.as_str().white() 
+            }, 
+            format!("{}", <std::time::SystemTime as Into<DateTime<Local>>>::into(std::fs::metadata(path.to_path_buf()).unwrap().modified().unwrap()).format("%b %d %Y %T")).dark_cyan(),
```

Comment:
Please figure out the proper way to do this, I believe @doinkythederp mentioned it in the Discord

Reply:
Yeah you should be able to just do this:
```suggestion
            format!("{}", DateTime::<Local>::from(std::fs::metadata(path.to_path_buf()).unwrap().modified().unwrap()).format("%b %d %Y %T")).dark_cyan(),
```

Reply:
Fixed in f560871